cmd/agent/app/settings/runtime_setting_dogstatsd_stats.go

```diff
@@ -0,0 +1,55 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package settings
+
+import (
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/cmd/agent/common"
+	"github.com/DataDog/datadog-agent/pkg/config"
+)
+
+// dsdStatsRuntimeSetting wraps operations to change log level at runtime.
+type dsdStatsRuntimeSetting string
+
+func (s dsdStatsRuntimeSetting) Description() string {
+	return "Enable/disable the dogstatsd debug stats. Possible values: true, false"
+}
+
+func (s dsdStatsRuntimeSetting) Name() string {
+	return string(s)
+}
+
+func (s dsdStatsRuntimeSetting) Get() (interface{}, error) {
+	return common.DSD.DebugMetricsStats, nil
+}
+
+func (s dsdStatsRuntimeSetting) Set(v interface{}) error {
+	var newValue bool
+
+	// to be cautious, take care of both calls with a string (cli) or a bool (programmaticaly)
+	str, ok := v.(string)
+	if !ok {
+		b, ok := v.(bool)
+		if !ok {
+			return fmt.Errorf("bad parameter provided")
+		}
+		newValue = b
+	} else {
+		switch str {
+		case "true":
+			newValue = true
+		case "false":
+			newValue = false
+		default:
+			return fmt.Errorf("bad parameter value provided: %v", str)
+		}
+	}
+
+	common.DSD.DebugMetricsStats = newValue
```

Comment:
@ogaca-dd There is: I thought about it while developing this and I've tried to spam the server to have the Go race detector notify of a race access, it didn't detect any but to me, it doesn't mean that they don't exist (maybe it doesn't detect them on boolean vars).
Because the `DebugMetricsStats` is in a hot path (the read part), I'm reluctant to use a mutex, and because this is not a critical value (we don't mind having it wrong for a few cycles) I've decided to push it that way.
I'll try to switch to an atomic int and benchmark if there is any performance impact, if not much, I'll consider use this instead. I'll keep you posted.

Reply:
I've switched to an atomic int as it doesn't seem that there is much performance impact with doing so.